index.js

```diff
@@ -152,7 +158,7 @@ function normalizeCorsOptions (opts) {
 }
 
 function addCorsHeadersHandler (fastify, options, req, reply, next) {
```

Comment:
I would change the `normalizeCorsOptions` to provide extra argument.

```
function normalizeCorsOptions (opts, dynamic) {
  /* same as before */
  // use || false to always set to boolean
  corsOptions.dynamic = dynamic || false
}
```

Reply:
it's neither normalization nor does the property capture all related dynamic-ness.

If it's what you want, I can make the change.

Reply:
Maybe change to a proper name?
Split into two function seems unnecessary as both of them is internal helper function.